.github/workflows/tests.yaml

```diff
@@ -126,7 +126,7 @@ jobs:
         strategy:
             matrix:
                 build_variant: [no-ble-tsan-clang, no-ble-asan-clang]
-                chip_tool: ["", -same-event-loop]
+                chip_tool: [""]
```

Comment:
This makes some sense for Darwin, where there is really always a separate event loop in practice.

For Linux, this is our only CI coverage for the "same event loop" mode, which is in fact the mode used by many, if not most, of our devices, no?  Why is it ok to drop the coverage there?

Reply:
If the variant of chip-tool that is used by most to test is the 'same-event-loop' variant, then I can enable that version for Linux instead, and omit testing the version that has different event loops. Would that suffice?

Reply:
No, I mean the actual matter stack running on many devices on the server side (not chip-tool) is the same-event-loop version.

The default chip-tool build is presumably not-same-event-loop.  So maybe we should run CI with just the same-event-loop variant and assume the other will be tested in manual testing?   Except the manual testing does not use asan//tsan....

Reply:
I guess maybe if all-clusters-app on "linux" in practice uses same-event-loop then it makes sense to stop testing it here, since that app will cover it...